### ISSUE TITLE
Complete example

### ISSUE THREAD

**Issue by luizcarvalho** [hellostealth/stealth#74]
**A good idea would be a complete sample of chatbot using Stealth**
Complete samples is a awesome way to start and learn fast.

Sample trivia, or weather reporter, or another.

PS: congratulations form this project this is fantastic, I'm very excited to use in my chatbot projects and help it grow.

**Comment by matthewblack** [comment]
Hey @luizcarvalho - Thanks for the kind words and we couldn’t agree more!

We’re actually working on that right now and I’ll keep you updated when we post our full examples.

**Comment by luizcarvalho** [comment]
Awesome... very very thanks. And congrats again.

**Comment by matthewblack** [comment]
Hey @luizcarvalho - We're still working on our examples. I'll ping you shortly when they're ready!  Thanks for being awesome.🥇

**Comment by luizcarvalho** [comment]
@matthewblack fantastic, thanks... I promises, I will be more patient :P, sorry. I'm very excited!! :laughing:

**Comment by luizcarvalho** [comment]
Heelloo friends,

I started a post serie about chatbot development with Stealth, we have just 2 posts in pt-BR. 
I'll give a code example of stealth project too =)

PART 1: https://medium.com/v%C3%ADdeos-de-ti/parte-1-stealth-crie-chatbots-incr%C3%ADveis-com-ferramentas-que-voc%C3%AA-conhece-e-adora-f46e1f9c6a5c

Part 2: https://medium.com/@luizcarvalho/desenvolvendo-seu-primeiro-chatbot-utilizando-o-framework-stealth-dbf1be1508c0

=)

**Comment by matthewblack** [comment]
@luizcarvalho - This is so awesome!! We'll make sure to tweet about it on our Twitter account.

**Comment by luizcarvalho** [comment]
Amazing =D

**Comment by narender2031** [comment]
hello, 
can we create our own service instead of facebook and twilio

**Comment by itsjbecks** [comment]
@narender2031 yes you can. I did this as we have our own client and interface we were using and I integrated stealth as a micro service.

You need to create your own service gem. I cloned the twilio example and modified it to connect to my existing API.

**Comment by dramalho** [comment]
Sorry to bump this up, but here we are. Stealth seems feature rich and it ticks a billion boxes for me, but I admit that it's lacking in concrete examples and AFAIK I can't exactly get a simple test mode, say open an irb console and "talk" to the bot.

Am I missing one huge trick or is this the case? 

Assuming otherwise, I'll look around, but I guess having at least *one* configured service is the minimum requirement to give Stealth a spin. What's the simplest possible service that you guys know of that I could look at and adapt for a quick prototype?

Thanks for reading :)

**Comment by mgomes** [comment]
@dramalho with the upcoming release of Stealth 2.0, we are planning on releasing some screencasts to take one through the full cycle of a bot. Unfortunately none of these are available now. We kind of just assumed Stealth 2.0 would be finished sooner and we didn't want to spend too much time making videos for an older Stealth version. Live and learn I guess.

As for a "simple" test service you can use via console, that is not something we offer. It's also not on the roadmap. Happy to something like that to our README if someone builds it, but I think building an entire chat service via console isn't something I personally need and isn't as simple as you suggest.

I'd recommend getting a Messenger or Twilio bot going for your testing. That's where it would be deployed anyway and so it's always best to test it on the platform your are developing for. Facebook Messenger is more feature rich, but the Twilio SMS route will be simpler for you to get going.

**Comment by dramalho** [comment]
> As for a "simple" test service you can use via console, that is not something we offer. It's also not on the roadmap. Happy to add something like that to our README if someone builds it, but I think building an entire chat service via console isn't something I personally need and isn't as simple as you suggest.

Right, I meant something as simple as (1) start a session (2) get a controller to just '.say(something)` and get replies synchronously .. as prototypes go that's probably fine and for dev work (or while offline, who doesn't to work from the plane right :) ) it would be fine

But the fact that you're telling me it's not as simple as I'm imagining is telling - at least of my ignorance ;)

So, unfair question, any timeframe for 2.0? How much of a re-haul is it, like to I hit that learning curve now or wait a bit for the new one?

**Comment by dramalho** [comment]
Sorry, like I said, I'm kidnapping this issue a little bit. I'm trying to get a feel from you guys on how I might move, once I come to grasp with things, I'm more than happy to give back in work too , not completely free loading here. 

But I admit between the readme and the doc site, it felt like something is missing, and in that sometimes it's easier to ask :)

(disclaimer and thanks)
